Issues with source tar ball

Our release tgz of the source code (`spicy.tar.gz`)has a couple of issues:

- It doesn't have a top-level `spicy` directory, unpacks directly into the current directory. It should unpack into `spicy-<version>/`.

- When building it reports:

```
/Users/robin/tmp/x/scripts/autogen-version --header /Users/robin/tmp/x/build/include/hilti/rt/autogen/version.h --git-root /Users/robin/tmp/x
Traceback (most recent call last):
  File "/Users/robin/tmp/x/scripts/autogen-version", line 155, in <module>
    major, minor, patch, *_ = version.split(".")
ValueError: not enough values to unpack (expected at least 3, got 1)
[16/338] Building CXX object 3rdparty/benchmark/src/CMakeFiles/benchmark.dir/benchmark_register.cc.o
ninja: build stopped: subcommand failed.
```

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -359,7 +359,7 @@ validate_release_tarball_task:
   unpack_script:
     - mkdir /tmp/spicy-tarball
     - tar xf spicy*.tar.gz -C /tmp/spicy-tarball
-    - cd /tmp/spicy-tarball
+    - cd /tmp/spicy-tarball/spicy-*
   configure_script:
     - ./configure --generator=Ninja --enable-ccache --prefix=/opt/spicy --with-zeek=/opt/zeek --enable-werror
   build_script:

--- a/scripts/autogen-version
+++ b/scripts/autogen-version
@@ -13,12 +13,18 @@ import tempfile
 
 
 def get_version(ref):
-    git_tree = subprocess.call(
+    if os.path.exists("VERSION"):
+        with open("VERSION", "r") as info:
+            hash = info.readline().split("=")[1].strip()
+            branch = info.readline().split("=")[1].strip()
+            git_version = info.readline().split("=")[1].strip()
+            return hash, branch, git_version
+
+    if subprocess.call(
         ["git", "rev-parse", "--git-dir"],
         stdout=subprocess.DEVNULL,
         stderr=subprocess.DEVNULL,
-    )
-    if git_tree == 0:
+    ) == 0:
         hash = subprocess.check_output(
             ["git", "rev-parse", "--short", ref], universal_newlines=True
         ).strip()
@@ -46,17 +52,9 @@ def get_version(ref):
             r"-([0-9]*)-g[0-9a-z]*", r".\1", re.sub(r"^v", "", raw_git_version)
         ).strip()
         return hash, branch, git_version
-    elif os.path.exists("VERSION"):
-        with open("VERSION", "r") as info:
-            hash = info.readline().split("=")[1].strip()
-            branch = info.readline().split("=")[1].strip()
-            git_version = info.readline().split("=")[1].strip()
-            return hash, branch, git_version
-    else:
-        print(
-            "Cannot extract version information: neither git repository nor VERSION file present"
-        )
-        exit(1)
+
+    print("Cannot extract version information: neither VERSION file nor git repository present")
+    exit(1)
 
 
 def write_if_changed(target, txt):

--- a/scripts/make-tarball.sh
+++ b/scripts/make-tarball.sh
@@ -4,7 +4,7 @@ set -e
 
 cd "$(git rev-parse --show-toplevel)" || exit 1
 
-TARBALL=spicy.tar
+TARBALL="${PWD}/spicy.tar"
 
 # Create archive for the main repo.
 rm -f "$TARBALL"
@@ -17,5 +17,15 @@ git submodule foreach --quiet 'cd $toplevel && tar rf '"${TARBALL}"' $sm_path'
 ./scripts/autogen-version --store VERSION
 tar rf "${TARBALL}" VERSION
 
-# Compress archive.
-gzip -9 "${TARBALL}"
+# Introduce a top-level `spicy` directory and compress the tarball.
+SCRATCH=$(mktemp -d)
+VERSION=$(./scripts/autogen-version --short)
+mkdir "${SCRATCH}/spicy-${VERSION}"
+(
+    cd "${SCRATCH}/spicy-${VERSION}" &&
+    tar xf "${TARBALL}" &&
+    rm -f "${TARBALL}" &&
+    cd .. &&
+    tar czf "${TARBALL}.gz" "spicy-${VERSION}"
+)
+rm -rf "${SCRATCH}"

